internal/services/logic/logic_app_integration_account_map_resource_test.go

```diff
@@ -0,0 +1,189 @@
+}
+
+func (r LogicAppIntegrationAccountMapResource) template(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-logic-%d"
+  location = "%s"
+}
+
+resource "azurerm_logic_app_integration_account" "test" {
+  name                = "acctest-ia-%d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+  sku_name            = "Basic"
+}
+`, data.RandomInteger, data.Locations.Primary, data.RandomInteger)
+}
+
+func (r LogicAppIntegrationAccountMapResource) basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_logic_app_integration_account_map" "test" {
+  name                     = "acctest-iamap-%d"
+  resource_group_name      = azurerm_resource_group.test.name
+  integration_account_name = azurerm_logic_app_integration_account.test.name
+  map_type                 = "Xslt"
+  content                  = file("testdata/integration_account_map_content.xsd")
+}
+`, r.template(data), data.RandomInteger)
+}
+
+func (r LogicAppIntegrationAccountMapResource) requiresImport(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_logic_app_integration_account_map" "import" {
+  name                     = azurerm_logic_app_integration_account_map.test.name
+  resource_group_name      = azurerm_logic_app_integration_account_map.test.resource_group_name
+  integration_account_name = azurerm_logic_app_integration_account_map.test.integration_account_name
+  map_type                 = azurerm_logic_app_integration_account_map.test.map_type
+  content                  = azurerm_logic_app_integration_account_map.test.content
+}
+`, r.basic(data))
+}
+
+func (r LogicAppIntegrationAccountMapResource) complete(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_logic_app_integration_account_map" "test" {
+  name                     = "acctest-iamap-%d"
+  resource_group_name      = azurerm_resource_group.test.name
+  integration_account_name = azurerm_logic_app_integration_account.test.name
+  map_type                 = "Xslt"
+  content                  = file("testdata/integration_account_map_content.xsd")
+
+  metadata = <<METADATA
+    {
+        "foo": "bar"
+    }
+METADATA
+}
+`, r.template(data), data.RandomInteger)
+}
+
+func (r LogicAppIntegrationAccountMapResource) update(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_logic_app_integration_account_map" "test" {
+  name                     = "acctest-iamap-%d"
+  resource_group_name      = azurerm_resource_group.test.name
+  integration_account_name = azurerm_logic_app_integration_account.test.name
+  map_type                 = "Xslt20"
+  content                  = file("testdata/integration_account_map_content2.xsd")
+
+  metadata = <<METADATA
+    {
+        "foo": "bar2"
+    }
+METADATA
```

Comment:
is there a reason we just don't use a map of strings for this?

Reply:
Because the type of this property is [`Metadata interface{}`](https://github.com/Azure/azure-sdk-for-go/blob/5133956b2975cea2033799ac5b4e16c6cb76101b/services/logic/mgmt/2019-05-01/logic/models.go#L2512)